rfq/negotiator.go

```diff
@@ -123,28 +123,28 @@ func (n *Negotiator) queryBidFromPriceOracle(peer route.Vertex,
 		ctx, assetId, assetGroupKey, assetAmount,
 	)
 	if err != nil {
-		return 0, 0, fmt.Errorf("failed to query price oracle for "+
+		return nil, 0, fmt.Errorf("failed to query price oracle for "+
 			"bid: %w", err)
 	}
 
 	// Now we will check for an error in the response from the price oracle.
 	// If present, we will convert it to a string and return it as an error.
 	if oracleResponse.Err != nil {
-		return 0, 0, fmt.Errorf("failed to query price oracle for "+
+		return nil, 0, fmt.Errorf("failed to query price oracle for "+
 			"bid price: %s", oracleResponse.Err)
 	}
 
 	// By this point, the price oracle did not return an error or a bid
 	// price. We will therefore return an error.
-	if oracleResponse.BidPrice == nil {
-		return 0, 0, fmt.Errorf("price oracle did not specify a bid " +
-			"price")
+	if oracleResponse.AssetRate.Coefficient.ToUint64() == 0 {
```

Comment:
> I think it can just compare the coefficient directly here?

Latest commit compare `BigInt` with `0`. So we need the `ToUint64`.

> Or if the value is nil if there's no response, then we can continue to rely on that. If it's nil, won't we trigger a nil pointer here if we always deref?

If `oracleResponse == nil` and `err != nil` then we have an internal coding error in the implementation of `n.cfg.PriceOracle.QueryBidPrice`. That return situation goes against a convention I thought we had. I'm not sure if I'm answering your question here.